lib/layout/data_grid.dart

```diff
@@ -270,6 +290,13 @@ class DataGridState extends WidgetState<DataGrid>
 
     _buildDataColumn(scopeManager);
     _buildChildren();
+
+    // Ensure that _parseVisibility runs only once as it is inside buildWidget. Cannot put it in initState
+    if (isFirstTimeRunning) {
```

Comment:
this optimization can cause issues if buildWidget is run again and in the meantime the row's visibility has changed. There is no need for this optimization, it can cause more issues than it can solve. Also the performance impact would be probably a few nano seconds at best